src/API/WP/NotificationsService.php

```diff
@@ -100,7 +118,7 @@ public function notify( string $topic, $item_id = null, $data = [] ): bool {
 		 * @param int $item_id The item_id for the notification.
 		 * @param string $topic The topic for the notification.
 		 */
-		if ( ! apply_filters( 'woocommerce_gla_notify', $this->is_ready() && in_array( $topic, self::ALLOWED_TOPICS, true ), $item_id, $topic ) ) {
+		if ( ! apply_filters( 'woocommerce_gla_notify', $this->is_ready( $this->get_datatype_from_topic( $topic ) ) && in_array( $topic, self::ALLOWED_TOPICS, true ), $item_id, $topic ) ) {
 			$this->notification_error( $topic, 'Notification was not sent because the Notification Service is not ready or the topic is not valid.', $item_id );
```

Comment:
This error is even harder to troubleshoot now. Because there can many conditions leading to this error.

Especially in a scenario where the connection is healthy and notifications are all working, but it just happens to be disabled for one service.

For example I get:
```
Error sending notification for Item ID 123 with topic coupon.create. Notification was not sent because the Notification Service is not ready or the topic is not valid.
```

I guess if we follow up in the connection page with showing exactly which topics are enabled it might be easier to troubleshoot (I assume that's coming in a later PR).

Reply:
Yes. It does